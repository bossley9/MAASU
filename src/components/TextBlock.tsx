import React, { FC, ReactNode } from 'react'
import { Block } from './Block'

type Props = {
  children: ReactNode
  title: string
  className?: string
}

export const TextBlock: FC<Props> = ({ children, title, className = '' }) => {
  return (
    <Block>
      <section className={className}>
        <h4 className="title4 tc my2">{title}</h4>
        {children}
      </section>
    </Block>
  )
}
