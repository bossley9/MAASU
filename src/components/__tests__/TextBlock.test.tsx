import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { TextBlock } from '../TextBlock'

describe('render', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<TextBlock title="header">hello</TextBlock>, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})

it('renders a matching snapshot', () => {
  const test = renderer.create(<TextBlock title="header">hello</TextBlock>)
  expect(test.toJSON()).toMatchSnapshot()
})
