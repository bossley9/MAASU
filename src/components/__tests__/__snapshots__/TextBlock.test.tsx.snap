// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a matching snapshot 1`] = `
<div
  className="my4 "
>
  <section
    className=""
  >
    <h4
      className="title4 tc my2"
    >
      header
    </h4>
    hello
  </section>
</div>
`;
